src/engine.ml

```diff
@@ -357,6 +357,7 @@ let maybe_kex_client rng config tls =
   if Tls.Engine.can_handle_appdata tls then
     let pre_master, random1, random2 = (rng 48, rng 32, rng 32) in
     Config.client_generate_connect_options config >>= fun options ->
+    let pull = Config.mem Pull config in
```

Comment:
`--client` implies `--tls-client` and `--pull` -- In Config we synthesize this. So this is fine.